Documentation.md

```diff
@@ -164,590 +174,606 @@ and made available to your JS client code.
+be called on behalf of the agent. There is only ever one agent per Streams
 instantiation and all contacts and actions are assumed to be taken on behalf of
 this one agent.
 
 ### `agent.onContactPending()`
-```
-agent.onContactPending((agent) { ... });
+```js
+agent.onContactPending(function(agent) { /* ... */ });
 ```
 Subscribe a method to be called whenever a contact enters the pending state for this particular agent.
 
 ### `agent.onRefresh()`
-```
-agent.onRefresh(function(agent) { ... });
+```js
+agent.onRefresh(function(agent) { /* ... */ });
 ```
 Subscribe a method to be called whenever new agent data is available.
 
 ### `agent.onStateChange()`
-```
-agent.onStateChange(function(agentStateChange) { ... });
+```js
+agent.onStateChange(function(agentStateChange) { /* ... */ });
 ```
 Subscribe a method to be called when the agent's state changes. The
 `agentStateChange` object contains the following properties:
 
-* `agent`: The Agent object.
-* `newState`: The name of the agent's new state.
+* `agent`: The `Agent` API object.
 * `oldState`: The name of the agent's previous state.
+* `newState`: The name of the agent's new state.
 
 ### `agent.onRoutable()`
-```
-agent.onRoutable(function(agent) { ... });
+```js
+agent.onRoutable(function(agent) { /* ... */ });
 ```
 Subscribe a method to be called when the agent becomes routable, meaning
 that they can be routed incoming contacts.
 
 ### `agent.onNotRoutable()`
-```
-agent.onNotRoutable(function(agent) { ... });
+```js
+agent.onNotRoutable(function(agent) { /* ... */ });
 ```
 Subscribe a method to be called when the agent becomes not-routable, meaning
 that they are online but cannot be routed incoming contacts.
 
 ### `agent.onOffline()`
-```
-agent.onOffline(function(agent) { ... });
+```js
+agent.onOffline(function(agent) { /* ... */ });
 ```
 Subscribe a method to be called when the agent goes offline.
 
 ### `agent.onError()`
+```js
+agent.onError(function(agent) { /* ... */ });
 ```
-agent.onError(function(agent) { ... });
-```
-Subscribe a method to be called when the agent is put into an error state.  This
+Subscribe a method to be called when the agent is put into an error state. This
 can occur if Streams is unable to get new agent data, or if the agent fails to
-accept an incoming contact, or in other error cases.  It means that the agent is
+accept an incoming contact, or in other error cases. It means that the agent is
 not routable, and may require that the agent switch to a routable state before
 being able to be routed contacts again.
 
 ### `agent.onSoftphoneError()`
-```
-agent.onSoftphoneError(function(agent) { ... });
+```js
+agent.onSoftphoneError(function(error) { /* ... */ });
 ```
 Subscribe a method to be called when the agent is put into an error state specific to softphone funcionality.
 
+The `error` argument is a `connect.SoftphoneError` instance with the following methods: `getErrorType()`, `getErrorMessage()`, `getEndPointUrl()`.
```

Comment:
Found out that this method does not pass an `Agent` object, but rather a `SoftphoneError` object, so decided to document it.

Reply:
Nice, thanks